cypress/support/TestListManager.js

```diff
@@ -3,225 +3,140 @@
+        it(test.name, test.func);
+        test = testListIterator.next;
       }
     })
   }
 }
 
-class TestListIterator
-{
-  constructor(testList)
-  {
-    this.testList = testList
-    this.index = 0
-    this.currentGroup = {name: ''}
+class TestListIterator {
+  constructor(testList) {
+    this.testList = testList;
+    this.index = 0;
+    this.currentGroup = {name: ''};
   }
 
   // groupName.testName - 'testName' from 'groupName'
   // TODO: Add support for these wild card versions
   // *.*                - All Groups, All Tests
   // *.testName         - All tests with all groups matching 'testName'
   // groupName.*        - All tests from 'groupName'
-  get next()
-  {
-    if (this.index >= this.testList.length) return undefined
+  get next() {
+    if (this.index >= this.testList.length) return undefined;
 
-    var x = this.testList[this.index].split('.')
-    if (x.length != 2) throw `Invalid item in testList[${this.index}]: "${this.testList[this.index]}" - 'group DOT testName' format is expected`
-    var groupName = x[0]
-    var testName = x[1]
+    var x = this.testList[this.index].split('.');
+    if (x.length != 2) throw `Invalid item in testList[${this.index}]: "${this.testList[this.index]}" - 'group DOT testName' format is expected`;
+    var groupName = x[0];
+    var testName = x[1];
 
-    if (this.currentGroup.name != groupName)
-    {
-      this.currentGroup = GetTestGroup(groupName)
-      if (this.currentGroup == undefined) throw `Group '${groupName}' NOT found in testGroups`
+    if (this.currentGroup.name != groupName) {
+      this.currentGroup = GetTestGroup(groupName);
+      if (this.currentGroup == undefined) throw `Group '${groupName}' NOT found in testGroups`;
     }
     
-    var test = GetTest(this.currentGroup, testName)
-    if (test == undefined) throw `Test '${testName}' NOT found in test group '${groupName}'`
+    var test = GetTest(this.currentGroup, testName);
+    if (test == undefined) throw `Test '${testName}' NOT found in test group '${groupName}'`;
 
-    this.index++
-    return {group: this.currentGroup.name, name: test.name, func: test.func}
+    this.index++;
+    return {group: this.currentGroup.name, name: test.name, func: test.func};
   }
 }
 
-function GetTestGroup(name)
-{
-  for (var i = 0; i < testGroups.length; i++)
-  {
-    if (testGroups[i].name == name) 
-    {
-      return testGroups[i]
-    }
+function GetTestGroup(name) {
+  for (var i = 0; i < testGroups.length; i++) {
+    if (testGroups[i].name == name) { return testGroups[i]; }
   }
-  return undefined
+  return undefined;
 }
 
-function GetTest(testGroup, testNameToFind)
-{
-  var toFind = `function ${testNameToFind}(`
-  for (var i = 0; i < testGroup.tests.length; i++)
-  {
-    if (`${testGroup.tests[i].func}`.startsWith(toFind)) 
-    {
-      return testGroup.tests[i]
+function GetTest(testGroup, testNameToFind) {
+  for (var i = 0; i < testGroup.tests.length; i++) {
```

Comment:
again use find and only pass minimal info to function. It doesn't need the whole `testGroup` only the `tests`

Example:
`tests.find(test => test.func.toString().substring(9).startsWith(testName))`

Reply:
Fixed. Thanks for pointing these out, being new to JavaScript there are still a lot of helpful functions and patterns that I don't know about.